core/state_processor.go

```diff
@@ -160,3 +163,23 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	vmenv := vm.NewEVM(blockContext, vm.TxContext{BlobHashes: tx.BlobHashes()}, statedb, config, cfg)
 	return applyTransaction(msg, config, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
 }
+
+// ProcessBeaconBlockRoot applies the EIP-4788 system call to the beacon block root
+// contract. This method is exported to be used in tests.
+func ProcessBeaconBlockRoot(beaconRoot common.Hash, vmenv *vm.EVM, statedb *state.StateDB) {
+	// If EIP-4788 is enabled, we need to invoke the beaconroot storage contract with
+	// the new root
+	msg := &Message{
+		From:      params.SystemAddress,
+		GasLimit:  100_000,
+		GasPrice:  new(big.Int).SetUint64(0), // TODO use nil?
+		GasFeeCap: new(big.Int).SetUint64(0), // TODO use basefee?
+		GasTipCap: new(big.Int).SetUint64(0), // TODO use zero?
+		To:        &params.BeaconRootsStorageAddress,
+		Data:      beaconRoot[:], // TODO use 4byte invocation?
```

Comment:
can probably remove the above `TODOs` above? I think setting zero is okay for these values. Is there a reason to not use `common.Big0` here though?